### PR TITLE
Use 0 as mid-point for AC in graph when feed-in is enabled

### LINKED ISSUE #1060: Brief page: Side panel grid graph
Shouldn't the graph for the grid start in the middle, if grid feed in is enabled?

grafik

Like this:

grafik

### PR DESCRIPTION
**Pull request by blammit**
Fixes #1060

### CODE REVIEW

File: pages/BriefSidePanel.qml [hunk 1]
```diff
@@ -166,7 +166,15 @@ exported power v  0.4 |   /
 				id: acInputGraphRange
 
 				phaseModel: root.visible ? Global.acInputs.activeInput.phases : null
-				minimumCurrent: Global.acInputs.activeInputInfo.minimumCurrent
+				minimumCurrent: {
+					if (Global.acInputs.activeInputInfo.minimumCurrent < 0
+							&& Global.acInputs.activeInputInfo.maximumCurrent > 0) {
+						// If the values may go below zero, then ensure that zero is the vertical
+						// mid-point of the graph.
+						return Global.acInputs.activeInputInfo.maximumCurrent * -1
```
Comment (chriadam) [inline pages/BriefSidePanel.qml:174]: what if the magnitude of minimumCurrent is far greater than that of maximumCurrent?  e.g. if minimumCurrent is -500 but maximumCurrent is 200?

Comment (blammit) [reply pages/BriefSidePanel.qml]: Oh, right -- have fixed this to use max(abs(min_current), max_current).

Comment (chriadam) [reply pages/BriefSidePanel.qml]: sounds good - the only issue that zero will no longer be the mid-point, but maybe that's not a problem?

Comment (chriadam) [reply pages/BriefSidePanel.qml]: Ah, I see, now you use that abs value as the max and min value.  Nice.

**Review by chriadam — APPROVED** [review]

**Comment by mr-manuel** [comment]
Somehow nothing changed on my end.

grafik

**Comment by mr-manuel** [comment]
I also think, that the graph doesn't have to be symmetrical. It can also be that the min is `-10A` and the max `30A`. This would result in the 0 mid-point to be at 1/4 from the bottom. Maybe this should be already like this?

**Comment by blammit** [comment]
> I also think, that the graph doesn't have to be symmetrical. It can also be that the min is -10A and the max 30A. This would result in the 0 mid-point to be at 1/4 from the bottom. Maybe this should be already like this?

That's how it should be working at the moment in `main`, yes.

I recall talking to Jesús about this previously, and he requested that 0 be the mid-point.

**Comment by mr-manuel** [comment]
I think I now understood the problem better. The mid-point is correct, but the graph fill is not. The graph fill should always be towards 0 and not the bottom. See https://github.com/victronenergy/gui-v2/issues/1060#issue-2300922125

**Comment by jpetrell** [comment]
I'll get these improvements merged for the release, the graph fill can be fixed separately.

**Pull request merged**